_drafts/wsl2-firmware-development.md

```diff
@@ -0,0 +1,655 @@
+  with an anecdote.
+  
+  As I was setting up my new laptop, I used my Microsoft
+  account when installing Windows, something that I had long avoided, not really
+  for any specific reason.  It felt like I could potentially avoid “Windows problems”
+  by being compliant here.  It went OK, except that it never prompted me for the
+  name of my home (AKA user) folder and then it named it incorrectly: “jphut” instead of
+  “jp”.  Now, many people would have left it there.  And perhaps I should have. 
+  But “jphut” is _wrong_.  Home is at `/home/jp`, or `c:/users/jp`, or `/users/jp` -
+  this is how it has _always been_.  “No big deal”, I thought. “I’ll fix it!” 
+  
+  It took me 3 hours.  RegEdit.  Reviewing scripts left on Windows forums. 
+  Seriously breaking my install.  Adding symlinks for OneDrive.  It’s been working
+  fine for a few months now, but for all I know I’m out of spec in a weird way that
+  will cause problems.  So, to quote Linus Torvalds, “F@&# you, \[Microsoft\]!”
+- 📎Unsettling paperclip mascot!
+
+#### VMWare Workstation Pro
+
+⭐⭐⭐
+
+
+Pros:
+- With a fully featured VM, you get most of the pros of Linux with access to Window’s strong points when you need them.
+- USB forwarding, including BLE, works well because you can run a full Linux distro of your choice with all the kernel modules that you need.
+- SSH to a headless VM allows for a snappy, low-latency, experience in your
+  editor of choice.
+
+Cons:
+- Setup and maintenance.  VMs are complicated, and setting up and maintaining the VM takes time and research.  A common example is the allocation of disk space.  Generally, you will need to allocate the space for your guest OS up front.  It can be hard to predict exactly how much space you really need, so you may end up needing to expand the virtual disk at some point, requiring steps to be taken from both the host and guest.
+
+#### WSL2
+
+⭐⭐⭐⭐
+
+Pros:
+- You get most of the pros of a Linux environment and all of the pros of the Windows host.
+- Resource allocation is automatic.  You can fiddle with CPU and RAM allocation, but you shouldn’t need to.  Storage is automatically resized as needed, so you never need to worry about running out of space for your WSL2 environment.
+- Setup is very simple — you can follow Intercreate’s tutorial below — and supported by Microsoft.
+
+Cons:
+- Minimal choice of distros and fewer options for customizing your installation and kernel.  That’s not to say that you _can’t_, but rather that you will stray outside of the Microsoft support umbrella by doing so.
+- BLE, and probably other kernel modules that I haven’t thought of, are not enabled in the default WSL2 configuration.
+- Performance is not as good as running Linux on bare metal.
+
+### Conclusion
+
+Ultimately, choosing a development environment depends on what helps you to
+provide the most value to your clients.  Decisions about security, performance,
```

Comment:
Are all Interrupt readers in the client-focused industry? For companies doing their own development it's about developer experience / velocity. Might want to delete 'to your clients'.

Reply:
Good point - maybe "product" - everyone in FW is working on something that can loosely be defined as a product , right?